File: Introduccion a Poo/Ejercicio 7/Python/Celular.py
class Aplicacion:
    def __init__(self, nombre, p):
        self.nombre = nombre
        self.p = p
class Celular:
    def __init__(self):
        self.e = 1024
        self.bateria = 100
        self.apps = []
    def intalar(self, app):
        if len(self.apps) < 20 and app.p <= self.e:
            self.apps.append(app)
            self.e = self.e - app.p
        else:
            print("no puedo instalarlo")
    def usar(self, nombre, minutos):
        if self.bateria <= 0:
            print("se apago :(")
            return
        for app in self.apps:
            if app.nombre == nombre:
                if app.p > 250:
                    gasto = 5
                elif app.p > 100:
                    gasto = 2
                else:
                    gasto = 1
                total = (minutos / 10) * gasto
                self.bateria = self.bateria - total
                if self.bateria <= 0:
                    self.bateria = 0
                    print("se apago :(")
                return
    def mostrar(self):
        print("Batería:", self.bateria, "%")
cel = Celular()
app1 = Aplicacion("Facebook", 300)
app2 = Aplicacion("Notas", 50)
cel.intalar(app1)
cel.intalar(app2)
cel.usar("Facebook", 30)
cel.mostrar()

File: Introduccion a Poo/Ejercicio 7/Java/Celular.java
import java.util.ArrayList;
public class Aplicacion {
    private String nombre;
    private int peso;
    public Aplicacion(String nombre, int peso) {
        this.nombre = nombre;
        this.peso = peso;
    }
    public String getNombre() {
        return nombre;
    }
    public int getPeso() {
        return peso;
    }
}
public class Celular {
    private ArrayList<Aplicacion> apps = new ArrayList<>();
    private int espacio = 1024;
    private double bateria = 100;
    public void instalar(Aplicacion app) {
        if (apps.size() < 20 && app.getPeso() <= espacio) {
            apps.add(app);
            espacio -= app.getPeso();
        } else {
            System.out.println("No se puede instalar");
        }
    }
    public void usar(String nombre, int minutos) {
        if (bateria <= 0) {
            System.out.println("Celular apagado");
            return;
        }
        for (Aplicacion app : apps) {
            if (app.getNombre().equals(nombre)) {
                double gasto;
                if (app.getPeso() > 250) {
                    gasto = 5;
                } else if (app.getPeso() > 100) {
                    gasto = 2;
                } else {
                    gasto = 1;
                }
                double total = (minutos / 10.0) * gasto;
                bateria -= total;
                if (bateria <= 0) {
                    bateria = 0;
                    System.out.println("Celular apagado");
                }
                return;
            }
        }
    }
    public void mostrar() {
        System.out.println("Batería restante: " + bateria + "%");
    }
}